compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1621,6 +1624,10 @@ public void visit(BLangSimpleVariableDef varDefNode) {
         analyzeDef(varDefNode.var, env);
     }
 
+    public void visit(BLangLetExpression letExpression) {
+        analyzeDef(letExpression, env);
+    }
+
```

Comment:
Do we need this?
If so, shall we visit letdeclarations and expressions with env of letexpression?
IE:
```java
   public void visit(BLangLetExpression letExpression) {
        for (BLangLetVariable letVarDeclaration : letExpression.letVarDeclarations) {
            analyzeDef((BLangNode) letVarDeclaration.definitionNode, letExpression.env);
        }
        analyzeDef(letExpression.expr, letExpression.env);
    }
```

Reply:
Seems it has been introduced with commit #7cb340695f485fb06251412f64e1c8fdff909720. Reverted it